internal/cloud/common.go

```diff
@@ -6,12 +6,15 @@ import (
 	"io"
 	"path/filepath"
 	"strings"
+
+	corev1 "k8s.io/api/core/v1"
 )
 
 type Common struct {
 	ClusterName       string     `env:"CLUSTER_NAME" validate:"required"`
 	ArtifactBucketURL *BucketURL `env:"ARTIFACT_BUCKET_URL,noinit" validate:"required"`
 	RegistryURL       string     `env:"REGISTRY_URL" validate:"required"`
+	Principal         string     `env:"PRINCIPAL"`
```

Comment:
I think this should be required (`validate:"required"`)?

Reply:
the reason I made this optional is because that would require us to always set it, even when you use something like kind or when you run the tests. I had it as `validate:"required"` at first but after running tests, I thought it would be better to make it optional.

Reply:
After refactoring the PR, I think this could be changed back to required. Do you think that's the right move here? Even though Kind won't need it? There might be other environments where workload identity isn't available.

Reply:
I see the benefit of requiring this (production environments need it) outweighing the negatives… dev envs needing to pass a stub value… alternative would be to have some custom validation logic, but prob not worth the effort there